x-pack/plugins/maps/public/shared/layers/sources/xyz_tms_source.js

```diff
@@ -13,12 +13,18 @@ import {
 
 import { AbstractTMSSource } from './tms_source';
 import { TileLayer } from '../tile_layer';
+import { i18n } from '@kbn/i18n';
+import { getDataSourceLabel, getUrlLabel } from '../../../../common/i18n_getters';
 
 export class XYZTMSSource extends AbstractTMSSource {
 
   static type = 'EMS_XYZ';
-  static title = 'Tile Map Service from URL';
-  static description = 'Map tiles from a URL that includes the XYZ coordinates';
+  static title = i18n.translate('xpack.maps.source.ems_xyz.title', {
```

Comment:
I'm not sure how this is consumed, but i guess this might be adhering more to the guidelines.
```suggestion
  static title = i18n.translate('xpack.maps.source.ems_xyzTitle', {
```